src/query/functions/tests/it/type_check/testdata/decimal.txt

```diff
@@ -2974,23 +2974,30 @@ evaluation (internal):
 +--------+-----------------------+
 
 
-error: 
-  --> SQL:1:6
-  |
-1 | d128 > '1'
-  |      ^ no function matches signature `gt(Decimal(38, 0), String)`, you might need to add explicit type casts.
-
-candidate functions:
-  gt(Variant, Variant) :: Boolean                               : unable to unify `Decimal(38, 0)` with `Variant`
-  gt(Variant NULL, Variant NULL) :: Boolean NULL                : unable to unify `Decimal(38, 0)` with `Variant`
-  gt(String, String) :: Boolean                                 : unable to unify `Decimal(38, 0)` with `String`
-... and 33 more
-
+ast            : d128 > '1'
+raw expr       : gt(d128::Decimal(38, 0), '1')
+checked expr   : gt<Float64, String>(CAST<Decimal(38, 0)>(d128 AS Float64), "1")
```

Comment:
Suggest checking why cast is not prioritized on the constant side here

Reply:
we don't implement decimal compare with string, so decimal value is auto casted to float.